src/pyfmi/fmi3.pyx

```diff
@@ -1941,6 +1973,201 @@ cdef class FMUModelBase3(FMI_BASE.ModelBase):
+                if ret != 0:
+                    raise FMUException(f"Unexpected failure retreiving dependencies of variable {output_var_name}")
+
+                if (numDependencies == 0) and (dependsOnAll == 0):
+                    states[output_var_name] = []
+                    states_kind[output_var_name] = []
+
+                    inputs[output_var_name] = []
+                    inputs_kind[output_var_name] = []
+                if (numDependencies == 0) and (dependsOnAll == 1):
+                    states[output_var_name] = states_list
+                    states_kind[output_var_name] = [FMI3_DependencyKind.DEPENDENT]*len(states_list)
+
+                    inputs[output_var_name] = inputs_list
+                    inputs_kind[output_var_name] = [FMI3_DependencyKind.DEPENDENT]*len(inputs_list)
+                else:
+                    states[output_var_name] = []
+                    states_kind[output_var_name] = []
+
+                    inputs[output_var_name] = []
+                    inputs_kind[output_var_name] = []
+                    for i in range(numDependencies):
+                        dependency_value_ref = dependencies[i]
+                        if dependency_value_ref in map_vr_to_state:
+                            states[output_var_name].append(map_vr_to_state[dependency_value_ref])
+                            states_kind[output_var_name].append(FMI3_DependencyKind(dependenciesKind[i]))
+                        elif dependency_value_ref in map_vr_to_input:
+                            inputs[output_var_name].append(map_vr_to_input[dependency_value_ref])
+                            inputs_kind[output_var_name].append(FMI3_DependencyKind(dependenciesKind[i]))
+                        else:
+                            pass # XXX: Not float64 or continuous
+
+        # Caching
+        self._outputs_states_dependencies = states
+        self._outputs_states_dependencies_kind = states_kind
+        self._outputs_inputs_dependencies = inputs
+        self._outputs_inputs_dependencies_kind = inputs_kind
+
+        return states, inputs
+
+    cpdef get_output_dependencies_kind(self):
+        """
+        Retrieve the 'kinds' that the (continuous, float64) outputs are
+        dependent on. Returns two dictionaries, one with the states
+        and one with the (continuous, float64) inputs. The list of 'kinds'::
+
+            FMI3_DependencyKind.DEPENDENT (= 0)
+            FMI3_DependencyKind.CONSTANT  (= 1)
+            FMI3_DependencyKind.FIXED     (= 2)
+            FMI3_DependencyKind.TUNABLE   (= 3)
+            FMI3_DependencyKind.DISCRETE  (= 4)
+
+        """
+        # TODO: More than just float64 outputs?
+        self.get_output_dependencies()
+        return self._outputs_states_dependencies_kind, self._outputs_inputs_dependencies_kind
+
+    cpdef get_derivatives_dependencies(self):
```

Comment:
Essentially copy paste of `get_output_dependencies()`, but with derivatives instead of outputs.